control-plane/csi-driver/src/bin/node/format.rs

```diff
@@ -1,46 +1,66 @@
 //! Utility function for formatting a device with filesystem
-use csi_driver::context::FileSystem;
+use crate::mount;
+use csi_driver::filesystem::FileSystem;
 
-use devinfo::blkid::probe::Probe;
-use std::process::Command;
-use tracing::{debug, trace};
+use tracing::debug;
+use uuid::Uuid;
 
-pub(crate) async fn prepare_device(device: &str, fstype: &FileSystem) -> Result<(), String> {
+/// Prepare the filesystem before mount, change parameters if requested.
+pub(crate) async fn prepare_device(
+    fstype: &FileSystem,
+    volume_uuid: &Uuid,
+    device: &str,
+    staging_path: &str,
+    options: &[String],
+    change_fs_id: bool,
+) -> Result<(), String> {
     debug!("Probing device {}", device);
+    let fs = FileSystem::get_property(device, "TYPE");
 
-    let probe =
-        Probe::new_from_filename(device).map_err(|error| format!("probe setup failed: {error}"))?;
+    let fs_ops = fstype.to_fs_ops()?;
 
-    if let Err(error) = probe.do_probe() {
-        return Err(format!("probe failed: {error}"));
-    }
-
-    if let Ok(fs) = probe.lookup_value("TYPE") {
+    if let Ok(fs) = fs {
         debug!("Found existing filesystem ({}) on device {}", fs, device);
+        if change_fs_id {
+            match fstype {
```

Comment:
Why not simply use the trait here?

Reply:
I was not able to put the `xfs_repair` in the trait because the mount module is not accessible in the FileSystem lib.